### ISSUE TITLE
[Bug] User interface freezes when clicking on function call in decompiler

### ISSUE THREAD

**Issue by 0xBEEEF** [NationalSecurityAgency/ghidra#5429]
**Describe the bug**
When clicking on a function to navigate to it, the user interface freezes for some functions. In this case, the entire interface is locked and no longer able to provide feedback. An exception occurs, which is mentioned below.

**To Reproduce**
It is difficult to recreate. Unfortunately I can't provide the binary file either, but with the old Ghidra 10.2.X version the error didn't occur. This was only added by the current version.

**Expected behavior**

**Screenshots**

**Attachments**
Attached is the occurring exception, which is only displayed in debug mode. Nothing is written in the error log itself. This is the only error I found during the analysis. If there is a possibility to find out more information here, please give me feedback what I have to set and how, so that the bug can be fixed.

```
ERROR EventManager Plugin Event Error: Error in plugin event listener
 ghidra.util.exception.ClosedException: File is closed
        at db.buffers.BufferMgr.getCachedBufferNode(BufferMgr.java:883)
        at db.buffers.BufferMgr.getBufferNode(BufferMgr.java:792)
        at db.buffers.BufferMgr.getBuffer(BufferMgr.java:956)
        at db.NodeMgr.getLongKeyNode(NodeMgr.java:257)
        at db.Table$LongKeyRecordIterator.<init>(Table.java:2083)
        at db.Table.iterator(Table.java:1868)
        at ghidra.program.database.map.AddressKeyRecordIterator.<init>(AddressKeyRecordIterator.java:146)
        at ghidra.program.database.map.AddressKeyRecordIterator.<init>(AddressKeyRecordIterator.java:62)
        at ghidra.program.database.code.DataDBAdapterV0.getRecordBefore(DataDBAdapterV0.java:108)
        at ghidra.program.database.code.CodeManager.getDefinedBefore(CodeManager.java:916)
        at ghidra.program.database.code.CodeManager.getCodeUnitBefore(CodeManager.java:958)
        at ghidra.program.database.ListingDB.getCodeUnitBefore(ListingDB.java:69)
        at ghidra.app.util.viewer.listingpanel.ProgramBigListingModel.getAddressBefore(ProgramBigListingModel.java:327)
        at ghidra.app.util.viewer.listingpanel.ListingModelAdapter.getIndexBefore(ListingModelAdapter.java:109)
        at docking.widgets.fieldpanel.internal.AnchoredLayoutHandler.getPreviousLayout(AnchoredLayoutHandler.java:194)
        at docking.widgets.fieldpanel.internal.AnchoredLayoutHandler.fillLayoutsBack(AnchoredLayoutHandler.java:183)
        at docking.widgets.fieldpanel.internal.AnchoredLayoutHandler.fillOutLayouts(AnchoredLayoutHandler.java:137)
        at docking.widgets.fieldpanel.internal.AnchoredLayoutHandler.shiftView(AnchoredLayoutHandler.java:112)
        at docking.widgets.fieldpanel.FieldPanel.scrollView(FieldPanel.java:119)
        at docking.widgets.fieldpanel.FieldPanel.goTo(FieldPanel.java:837)
        at docking.widgets.fieldpanel.FieldPanel.goTo(FieldPanel.java:821)
        at ghidra.app.util.viewer.listingpanel.ListingPanel.goTo(ListingPanel.java:589)
        at ghidra.app.plugin.core.codebrowser.CodeViewerProvider.setLocation(CodeViewerProvider.java:696)
        at ghidra.app.plugin.core.codebrowser.CodeBrowserPlugin.processEvent(CodeBrowserPlugin.java:130)
        at ghidra.framework.plugintool.Plugin.eventSent(Plugin.java:293)
        at ghidra.framework.plugintool.mgr.EventManager.sendEvents(EventManager.java:286)
        at ghidra.framework.plugintool.mgr.EventManager.lambda$new$3(EventManager.java:49)
        at ghidra.util.Swing.doRun(Swing.java:292)
        at ghidra.util.Swing.runNow(Swing.java:208)
        at ghidra.util.Swing.runNow(Swing.java:163)
        at ghidra.framework.plugintool.mgr.EventManager.fireEvent(EventManager.java:216)
        at ghidra.framework.plugintool.PluginTool.firePluginEvent(PluginTool.java:511)
        at ghidra.framework.plugintool.Plugin.firePluginEvent(Plugin.java:456)
        at ghidra.app.plugin.core.gotoquery.GoToServicePlugin$DefaultNavigatable.goTo(GoToServicePlugin.java:123)
        at ghidra.app.plugin.core.gotoquery.GoToHelper.doGoTo(GoToHelper.java:120)
        at ghidra.app.plugin.core.gotoquery.GoToHelper.lambda$goTo$0(GoToHelper.java:84)
        at ghidra.util.Swing.lambda$runNow$1(Swing.java:148)
        at ghidra.util.Swing.doRun(Swing.java:292)
        at ghidra.util.Swing.runNow(Swing.java:208)
        at ghidra.util.Swing.runNow(Swing.java:163)
        at ghidra.util.Swing.runNow(Swing.java:148)
        at ghidra.app.plugin.core.gotoquery.GoToHelper.goTo(GoToHelper.java:84)
        at ghidra.app.util.navigation.GoToServiceImpl.goTo(GoToServiceImpl.java:61)
        at ghidra.app.plugin.core.decompile.DecompilerProvider.goToFunction(DecompilerProvider.java:652)
        at ghidra.app.decompiler.component.DecompilerController.goToFunction(DecompilerController.java:283)
        at ghidra.app.decompiler.component.DecompilerPanel.tryGoToFunction(DecompilerPanel.java:757)
        at ghidra.app.decompiler.component.DecompilerPanel.tryToGoto(DecompilerPanel.java:722)
        at ghidra.app.decompiler.component.DecompilerPanel.buttonPressed(DecompilerPanel.java:701)
        at docking.widgets.fieldpanel.FieldPanel.lambda$notifyFieldMouseListeners$0(FieldPanel.java:1293)
        at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
        at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:771)
        at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:722)
        at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:716)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
        at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
        at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:741)
        at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
```

**Environment (please complete the following information):**
 - OS: Windows 11
 - Java Version: 17.0.7.7 Amazon Corretto
 - Ghidra Version: 10.3
 - Ghidra Origin: official GitHub distro

**Additional context**

**Comment by 0xBEEEF** [comment]
Great news! The bug was apparently fixed with the recently released patch 10.3.1. For this reason, I am closing this open issue. It is not possible for me to reproduce the behavior with it. Should I discover the bug again in a different form, I will simply open a new issue.

**Comment by ryanmkurtz** [comment]
I think this was a duplicate of #5400.
